src/modules/job-manager/job-manager.c

```diff
@@ -105,6 +132,10 @@ int mod_main (flux_t *h, int argc, char **argv)
         flux_log_error (h, "flux_reactor_run");
         goto done;
     }
+    if (checkpoint_to_kvs (&ctx) < 0) {
```

Comment:
Perhaps we need a way for services that do checkpoints to determine if the current instance is running with a backing store that will be persistent. If not, then checkpoint is fruitless, and could be skipped for faster shutdown.

For now it probably doesn't matter, but when we have a lot of services all doing checkpoints at shutdown, it could add up.

(apologies if this is done already and I just missed it!)

Reply:
Good thought - I was also wondering if we should establish a protocol for where these go in the KVS, and what they contain.  Dates might be useful for example.  An event to trigger writing of checkpoints might be useful.  Should we write them out periodically in case instance dies?

Larger topic than this PR I think.